Examples/Nodify.Shapes/App.xaml

```diff
@@ -0,0 +1,226 @@
+             xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
+             xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+             StartupUri="MainWindow.xaml">
+    <Application.Resources>
+        <ResourceDictionary>
+            <Style x:Key="IconButton"
+                   TargetType="Button"
+                   BasedOn="{StaticResource {x:Type Button}}">
+                <Setter Property="Background"
+                        Value="Transparent" />
+                <Setter Property="BorderBrush"
+                        Value="Transparent" />
+                <Setter Property="Padding"
+                        Value="2" />
+                <Setter Property="Cursor"
+                        Value="Hand" />
+                <Setter Property="Template">
+                    <Setter.Value>
+                        <ControlTemplate TargetType="{x:Type Button}">
+                            <Border x:Name="PART_Border"
+                                    BorderThickness="{TemplateBinding BorderThickness}"
+                                    Background="{TemplateBinding Background}"
+                                    Padding="{TemplateBinding Padding}"
+                                    CornerRadius="3"
+                                    SnapsToDevicePixels="True">
+                                <ContentPresenter x:Name="contentPresenter"
+                                                  Focusable="False"
+                                                  Margin="{TemplateBinding Padding}"
+                                                  SnapsToDevicePixels="{TemplateBinding SnapsToDevicePixels}"
+                                                  HorizontalAlignment="{TemplateBinding HorizontalContentAlignment}"
+                                                  VerticalAlignment="{TemplateBinding VerticalContentAlignment}" />
+                            </Border>
+                        </ControlTemplate>
+                    </Setter.Value>
+                </Setter>
+                <Style.Triggers>
+                    <Trigger Property="IsEnabled"
+                             Value="False">
+                        <Setter Property="Opacity"
+                                Value="0.4" />
+                    </Trigger>
+                </Style.Triggers>
+            </Style>
+
+            <!--ICONS-->
+
+            <Viewbox x:Key="MinusIcon"
+                     Stretch="Fill">
+                <Grid>
+                    <Path StrokeLineJoin="Round"
+                          StrokeStartLineCap="Round"
+                          StrokeEndLineCap="Round"
+                          Data="M0 0h24v24H0z" />
+                    <Path Stroke="White"
+                          StrokeLineJoin="Round"
+                          StrokeStartLineCap="Round"
+                          StrokeEndLineCap="Round"
+                          Data="M5 12h14" />
+                </Grid>
+            </Viewbox>
```

Comment:
Defining controls in Resources is not the best practise, is it? This works in WPF, but i.e. only one control can use this resource, as a single control can't be attached to two parents. 

How about wrapping it in a `DataTemplate` and then binding `ContentTemplate` instead of `Content`? Alternatively,`x:Shared="False"` attribute can be added and technically this should work just fine (unfortunately not supported in [avalonia yet](https://github.com/AvaloniaUI/Avalonia/issues/15518), but I can always use the first approach in the port)

Reply:
Fixed using `DataTemplate`. Thanks!